### PR TITLE
feat: Add `regenerator‑runtime` type definition

### PR DESCRIPTION
**Pull request by ExE-Boss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/regenerator-runtime/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    tryLocsList?: TryLocationsList,
+    PromiseImpl?: ResolvablePromiseConstructorLike,
+): AsyncIterator<
+    TYield extends PromiseLike<infer Await> ? Await : Exclude<TYield, awrap<unknown>>,
+    TReturn extends PromiseLike<infer Await> ? Await : Exclude<TReturn, awrap<unknown>>
+>;
+export function async<T = undefined, TReturn = unknown>(
+    innerFn: InnerFunction<T, unknown, TReturn>,
+    // tslint:disable-next-line: ban-types
+    outerFn?: Function | null,
```
Comment (sheetalkamat) [inline types/regenerator-runtime/index.d.ts:211]: Do not disable this error:: Fix this instead by changing to `() => void`

File: types/regenerator-runtime/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+         * Expects properties to match `/^t[+-]?\d*(?:(?<=\d)\.\d*|\.\d+)?(?:e[+-]?\d+)?$/`.
+         */
+        temp: string
+    ]: any;
+}
+
+export function wrap<T = undefined, TYield = unknown, TReturn = unknown, TNext = unknown>(
+    innerFn: InnerFunction<T, TYield, TReturn, TNext>,
+    // tslint:disable-next-line: ban-types
+    outerFn?: Function | null,
```
Comment (sheetalkamat) [inline types/regenerator-runtime/index.d.ts:172]: Same fix this error

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/regenerator-runtime/index.d.ts (hunk 1 above)
Comment (ExE-Boss) [reply types/regenerator-runtime/index.d.ts:211]: That signature is incorrect, as it won’t accept class constructors and anything that takes at least one required parameter.

The `Function` type most closely describes what the `wrap` and `async` functions need: <https://github.com/facebook/regenerator/blob/0566ce53c67b5fe1de13b47ed4db00570de7d9f6/packages/regenerator-runtime/runtime.js#L39>.

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by ExE-Boss** [comment]
Ready to merge

**Pull request merged**